proxy/http/Http1Transaction.h

```diff
@@ -26,110 +26,43 @@
 #include "../ProxyTransaction.h"
 
 class Continuation;
+class Http1ClientSession;
 
 class Http1Transaction : public ProxyTransaction
 {
 public:
   using super_type = ProxyTransaction;
 
   Http1Transaction() {}
-  // Implement VConnection interface.
-  VIO *
-  do_io_read(Continuation *c, int64_t nbytes = INT64_MAX, MIOBuffer *buf = nullptr) override
-  {
-    return proxy_ssn->do_io_read(c, nbytes, buf);
-  }
-  VIO *
-  do_io_write(Continuation *c = nullptr, int64_t nbytes = INT64_MAX, IOBufferReader *buf = nullptr, bool owner = false) override
-  {
-    return proxy_ssn->do_io_write(c, nbytes, buf, owner);
-  }
-
-  void
-  do_io_close(int lerrno = -1) override
-  {
-    proxy_ssn->do_io_close(lerrno);
-    // this->destroy(); Parent owns this data structure.  No need for separate destroy.
-  }
-
-  // Don't destroy your elements.  Rely on the Http1ClientSession to clean up the
-  // Http1Transaction class as necessary.  The super::destroy() clears the
-  // mutex, which Http1ClientSession owns.
-  void
-  destroy() override
-  {
-    current_reader = nullptr;
-  }
-
-  void
-  do_io_shutdown(ShutdownHowTo_t howto) override
-  {
-    proxy_ssn->do_io_shutdown(howto);
-  }
-
-  void
-  reenable(VIO *vio) override
-  {
-    proxy_ssn->reenable(vio);
-  }
-
-  void
-  set_reader(IOBufferReader *reader)
-  {
-    sm_reader = reader;
-  }
 
+  ////////////////////
+  // Methods
+  void init_transaction(Http1ClientSession *ssn); // overload, not override
```

Comment:
I'm fine with overloading, but why do we want this method? I think we can pass Http(1|2|3)ClientSesion to constructors.

Reply:
Ah, I read the commit message. You're going to make constructors. +1.